editor/src/clj/internal/util.clj

```diff
@@ -546,3 +558,94 @@
    (deep-keep-kv-helper identity value-fn nil value))
   ([finalize-coll-value-fn value-fn value]
    (deep-keep-kv-helper finalize-coll-value-fn value-fn nil value)))
+
+(defn into-multiple
+  "Like core.into, but transfers the input into multiple collections in a single
+  pass. Instead of a single destination and optional transducer, takes a seq of
+  destinations and a seq of transducers. The number of destinations must match
+  the number of transducers. Returns a seq of the populated destinations."
+  ([] [])
+  ([tos] tos)
+  ([tos from]
+   (if (empty? from)
+     tos
+     (case (long (bounded-count 3 tos))
+       ;; No destinations - return unaltered.
+       0 tos
+
+       ;; Single destination - use regular into.
+       1 [(into (first tos) from)]
+
+       ;; Optimized for two destinations.
+       2 (let [transient-tos
+               (reduce (fn [transient-tos item]
+                         (pair (conj! (transient-tos 0) item)
+                               (conj! (transient-tos 1) item)))
+                       (pair (transient (nth tos 0))
+                             (transient (nth tos 1)))
+                       from)]
+           (pair (with-meta (persistent! (transient-tos 0)) (meta (nth tos 0)))
+                 (with-meta (persistent! (transient-tos 1)) (meta (nth tos 1)))))
+
+       ;; General case.
+       (mapv #(with-meta (persistent! %2) (meta %1))
+             tos
+             (reduce (fn [transient-tos item]
+                       (mapv conj! transient-tos (repeat item)))
+                     (mapv transient tos)
+                     from)))))
+  ([tos xforms from]
```

Comment:
Maybe add a precondition check that the count of tos matches the count of xforms?